plugins/agents/ping/Ping.go

```diff
@@ -0,0 +1,113 @@
+package ping
+
+import (
+	"strings"
+	"time"
+
+	"github.com/abrander/agento/plugins"
+	"github.com/abrander/agento/timeseries"
+	gping "github.com/gansoi/gansoi/plugins/agents/ping"
+)
+
+type Data struct {
+	Loss    int    `json:"loss"`
+	Sent    int    `json:"sent"`
+	Replies int    `json:"replies"`
+	TimeAvg int64  `json:"timeavg"`
+	TimeMin int64  `json:"timemin"`
+	TimeMax int64  `json:"timemax"`
+	Ip      string `json:"server"`
+}
+
+type Ping struct {
+	Data   []Data `json:"data"`
+	pinger *gping.ICMPService
+
+	Ip    string `toml:"ip" json:"ip" description:"The ip(s) to ping (multiple can be separated by comma)"`
+	Count int    `toml:"count" json:"count" description:"Number of packages to send"`
+}
+
+func init() {
+	plugins.Register("ping", NewPing)
+}
+
+func NewPing() interface{} {
+	return new(Ping)
+}
+
+func (p *Ping) Gather(transport plugins.Transport) error {
+	// Create and start the ICMP pinger service
+	if p.pinger == nil {
```

Comment:
This is not thread safe. I would suggest making `pinger` package level and initializing in `init()`. That would also save some time on each gathering.

Reply:
@abrander done